lib/dl_api_lib/dl_api_lib/query/formalization/query_formalizer.py

```diff
@@ -504,6 +506,19 @@ def make_relation_and_avatar_specs(
             for avatar_id in field_ava_dep_mgr.get_field_avatar_references(field_id)
         }
 
+        # Add avatars that participate in required relations
+        avatar_ids_by_required_relations: set[AvatarId] = functools.reduce(
```

Comment:
i strongly oppose using `reduce` here. i think the following can be written using nested comprehensions:

`avatar_ids_by_required_relations = {avatar_id for relation in self._ds_accessor.get_avatar_relation_list() for avatar_id in (relation.left_avatar_id, relation.right_avatar_id) if relation.required}`

or just keep the for loop implementation 5head

Reply:
Teach me, what's so bad about reduce?

Reply:
Discussed: reduce is not justified here as it is very much not readable (I for one have harder time reading double `for`s than reduce, but I see why reduce can also be harder to read than it should be)